### DISCUSSION TITLE
Logging blocked network request

### DISCUSSION THREAD

**Discussion by pandalamp** [sandboxie-plus/Sandboxie#1916] [Various ideas and suggestions]
When using a sandbox without internet privileges, I receive a single otification when the first internet access attempt is blocked. Is there a way to see the destination & port of that attempt, or further to log all such requests in a session?

Or instead perhaps is there a way to just prompt approval for each request alongside its destination?

**Comment by DavidXanatos** [comment]
No this level of fine grain interactive blocking is currently not supported, 
it could be added though if there is enough interest in such a feature.

**Reply by pandalamp** [reply]
> No this level of fine grain interactive blocking is currently not supported, it could be added though if there is enough interest in such a feature.

Thank you for your quick reply and consideration. Absent a granular implementation, please consider at minimum logging network requests attempted when a sandbox disallows them. That alone would be very useful to help evaluate if a program might be behaving maliciously or is just checking for updates.

For example, instead of "FILE.exe (PID): SBIE1307 Program cannot access the Internet due to restrictions," log something to the Sbie Messages pane or Notifications window like: "FILE.exe (PID): Restrictions blocked TCP connection to https://github-enterprise.s3.amazonaws.com/release/latest.json".

**Comment by DavidXanatos** [comment]
enable this:
![grafik](https://user-images.githubusercontent.com/3890945/171673782-60a67104-ffc9-466b-8460-c3f6102acdcb.png)

and that:

![grafik](https://user-images.githubusercontent.com/3890945/171674098-6f6a79d9-b376-40b7-b5d3-9af8c931701e.png)

and enable and watch the trace log

**Reply by pandalamp** [reply]
Thank you for your response and screenshots.

I selected the options you illustrated and have not found anything meaningful from the trace log.
![tracelog](https://user-images.githubusercontent.com/65859489/177411087-ba3437fd-87c8-45c2-80b6-0cdfbd4ff53c.jpg)

Searching through the larger log ("Status: [All]") also didn't seem to show what I was looking for either: I was just hoping to see the specific network addresses that were blocked when Sandboxie's error message was triggered.
